### ISSUE TITLE
Provide STORE support for ANNOTATION (RFC 5257), X-GM-LABELS (gmail extension), etc.

### ISSUE THREAD

**Issue by asutherland** [whiteout-io/browserbox#63]
STORE-command-wise, browserbox currently has setFlags(sequence, flags, options, callback) which assumes FLAGS.

Gmail supports an additional (proprietary) type, X-GM-LABELS (https://developers.google.com/gmail/imap_extensions#access_to_gmail_labels_x-gm-labels).  They give the example of:
```
a011 STORE 1 +X-GM-LABELS (foo)
* 1 FETCH (X-GM-LABELS (\Inbox \Sent Important "Muy Importante" foo))
a011 OK STORE (Success)
```

RFC 5257 (https://tools.ietf.org/html/rfc5257) has an example of:
```
C: a STORE 1 ANNOTATION (/comment (value.priv "My new comment"))
S: a OK Store complete
```

Given that most clients only need flags support and for backwards compatibility, it makes sense for setFlags to stay around.  But perhaps a more generic store() command could be added that follows the lead of setFlags.  Ex:
```js
conn.store(
  uidSet,
  {
    'X-GM-LABELS': {
      add / remove / set: ['foo', 'bar']
    }
  },
  { byUid: true });
```

Of course, one wrinkle is that a dictionary implies you can issue X-GM-LABELS and an ANNOTATION and maybe FLAGS all at the same time.  But STORE of course only can do one thing per tag, and pretending otherwise complicates error handling.  So store() could alternately be more low level, like:
```js
store(uidSet, '+X-GM-LABELS', ['foo', 'bar'], { byUid: true });
```
The value argument that holds the labels to add would be transformed into imap-handler tagged types by default, but the options dict could take something like "raw" to allow it to be passed through to imap-handler directly.

And the other other option is just to create custom methods for each STORE type: setGmailLabels, setAnnotation, etc.

**Comment by andris9** [comment]
Adding generic STORE makes a lot of sense. I wouldn't touch the METADATA stuff (including the ANNOTATION) at all at least until major IMAP servers fully support it which isn't happening any time soon, so I do not care too much of it. X-GM-LABELS is very similar to FLAGS, so it isn't an issue to add it.

**Comment by asutherland** [comment]
Awesome!  Thank you!  I think it does make sense to close this issue since I don't actually care about the METADATA stuff yet, I was just including it for discussion purposes.  If/when I need that I'll open a new issue.  (So, closing now.)

**Comment by asutherland** [comment]
Whoops, and I didn't realize your commit was still on a branch.  But assuming that's going to eventually get merged in to master after review or whatever, there's no real reason to keep this issue open, but I'll leave it to you to reopen if you want it that way for tracking.

**Comment by andris9** [comment]
I intended to keep the issue open at first as Github would have closed the issue automatically once the PR is merged but I accidentally hit Close and that broke the link between the issue and the PR, so it doesn't really matter anymore if the issue is open or closed.
